File: mycollab-mobile/src/main/java/com/mycollab/mobile/ui/AbstractMobileMainView.java
package com.mycollab.mobile.ui;

import com.mycollab.vaadin.mvp.PageView;
import com.mycollab.vaadin.mvp.ViewEvent;
import com.vaadin.ui.VerticalLayout;

import java.io.Serializable;

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
public abstract class AbstractMobileMainView extends VerticalLayout implements PageView, Serializable {
    private static final long serialVersionUID = 1L;

    @Override
    public void addViewListener(ViewListener listener) {
        addListener(ViewEvent.VIEW_IDENTIFIER(), ViewEvent.class, listener, ViewListener.viewInitMethod);
    }
}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/ui/AbstractMobileMainView.kt
package com.mycollab.mobile.ui

import com.mycollab.vaadin.events.ViewEvent
import com.mycollab.vaadin.mvp.PageView
import com.vaadin.ui.VerticalLayout
import java.io.Serializable

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
abstract class AbstractMobileMainView : VerticalLayout(), PageView {
    override fun <E> addViewListener(listener: PageView.ViewListener<E>) {
        addListener(ViewEvent.VIEW_IDENTIFIER, ViewEvent::class.java, listener, PageView.ViewListener.viewInitMethod)
    }
}